src/azure-cli/azure/cli/command_modules/acs/managed_cluster_decorator.py

```diff
@@ -3916,6 +3957,21 @@ def _restore_defaults_in_mc(self, mc: ManagedCluster) -> ManagedCluster:
                 setattr(mc, key, value)
         return mc
 
+    def set_up_defender(self, mc: ManagedCluster) -> ManagedCluster:
```

Comment:
Please also add some unit tests for `get_defender_config`, `set_up_defender`, `update_defender`.

May refer to the examples [test_get_node_resource_group](https://github.com/tzachiabo/azure-cli/blob/zaabow/AddDefenderSupport/src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py#L3772), [test_set_up_node_resource_group](https://github.com/tzachiabo/azure-cli/blob/zaabow/AddDefenderSupport/src/azure-cli/azure/cli/command_modules/acs/tests/latest/test_managed_cluster_decorator.py#L5568).